### ISSUE TITLE
Viber support

### ISSUE THREAD

**Issue by kerozoli** [bil-elmoussaoui/Hardcode-Tray#164]
Latest viber fully supports ubuntu-tray and viber-wrapper not working properly anymore.

Here's the new viber tray icon without viber-wrapper:
![viber](https://cloud.githubusercontent.com/assets/10573944/15649789/6496bc16-2674-11e6-8ad1-54512de6713a.png)

Could you please support Viber ?
(I could work on the support if necessary but unfortunetly only after June 20)

Thanks

**Comment by bil-elmoussaoui** [comment]
We will see what we can do about it :)
And btw, we do support Nylas too. You can request the icons in Numix theme if you want that :)
The icon names are 
- `nylas-tray-full-newitems`
- ` nylas-tray-full`
- `nylas-tray-empty`

**Comment by wa4557** [comment]
Since viber is not open source but proprietary, I'm not sure 
a) if we can do that, because the software is only distributed as a binary blob, which makes it very hard to unpack it / change it
b) if we are allowed to do that. I have no idea about licencing but changing the tray icon, might be violating the licence agreement

**Comment by bil-elmoussaoui** [comment]
@wa4557 I was thinking of forking the old viber wrapper indicator, port it to python3 and make it use the icon names provided by the default themes. And add that to `/database/scripts/` once the user install it, the script is copied to `/usr/bin/` as `viber-wrapper` and it changes the desktop file to execute `viber-wrapper` instead of `viber`

**Comment by wa4557** [comment]
Actually fixing the app should be fairly easy,because there seems to be an icon folder where we can replace the icons. As I said I'm not sure if we are allowed to do that; but I actually think it's fine

**Comment by bil-elmoussaoui** [comment]
@wa4557 Already tried that, the application does not use the icons in `/icons` folder :(

**Comment by wa4557** [comment]
Ah ok, if vber-wrapper provides the same functionality as the original tray icon, then I think your idea is pretty cool :)

**Comment by bil-elmoussaoui** [comment]
Yeah, the tray icon only provides `Open Viber` and `Exit`. I will create a branch, with the old script from `Viber-Wrapper-Indicator` repo and start tweaking it ;)

**Comment by bil-elmoussaoui** [comment]
@wa4557  I've pushed a first commit to port the code to Gtk3, still need some work. I might need some help if you're free of course

**Comment by wa4557** [comment]
Sure, what should I do?

**Comment by bil-elmoussaoui** [comment]
I've updated the code. Still few `os.system` that should be replaced by `subproccess`. Other than that, the code is on Gtk3 now, but runing it shows an exception 
```
Using OLD detection method (INTERNAL)
Launching Viber (/opt/viber/Viber) ... OK
Viber process terminated. Quitting...
Exiting because of exception 'WindowIsNone: Window is None'
```
Something still need to be updated but i don't have the time to look at the moment

**Comment by wa4557** [comment]
I'll have a look. I am a bit busy myself right now...

Bilal Elmoussaoui <notifications@github.com> schrieb am Fr., 3. Juni 2016,
16:42:

**Comment by wa4557** [comment]
Yeah, the script seems to be not working and not detecting a running viber instance. The rewrite of viber also seems to have broken viberwrapper-indicator (as far as I understand, I might be wrong). 

This almost makes it impossible to use the already existiing script imo

**Comment by bil-elmoussaoui** [comment]
We should just rewrite the detection part of the script. The indicator and actions(hide/show and quit) should work without any problem. I will try to find a way to replace the old code used to detect the runing process

**Comment by BlueManCZ** [comment]
On Gnome with [patched appindicator extension](https://github.com/bilelmoussaoui/Hardcode-Tray/files/5008706/appindicatorsupport%40rgcjonas.gmail.com.zip) and brightness set to 1 Viber icon looks great.

![Snímek z 2020-08-01 10-34-08](https://user-images.githubusercontent.com/17854950/89097845-f550bc00-d3e2-11ea-9220-213be97ba3bb.png)
![Snímek z 2020-08-01 10-34-31](https://user-images.githubusercontent.com/17854950/89097847-f84bac80-d3e2-11ea-99ed-63630e5e0228.png)

**Comment by rauldipeas** [comment]
Any news on this thread? Any chances to fix Viber?

![Screenshot_20221226_010931](https://user-images.githubusercontent.com/12108031/209498732-e67b0bca-5b5f-4fc8-a165-6c9f44100e54.png)
